src/BackgroundTask/CronBasedBackgroundTask.php

```diff
@@ -0,0 +1,191 @@
+		self::DEFAULT_INTERVAL_TWICE_DAILY,
+		self::DEFAULT_INTERVAL_DAILY,
+	];
+
+	/**
+	 * Name of the plugin as WordPress is expecting it.
+	 *
+	 * This should usually have the form "amp/amp.php".
+	 *
+	 * @var string
+	 */
+	private $plugin_file;
+
+	/**
+	 * Register the service with the system.
+	 *
+	 * @return void
+	 */
+	public function register() {
+		add_action( 'admin_init', [ $this, 'schedule_event' ] );
+		add_action( $this->get_event_name(), [ $this, 'process' ] );
+
+		$this->plugin_file = plugin_basename( dirname( dirname( __DIR__ ) ) . '/amp.php' );
+		add_action( "network_admin_plugin_action_links_{$this->plugin_file}", [ $this, 'add_warning_sign_to_network_deactivate_action' ], 10, 1 );
+		add_action( 'plugin_row_meta', [ $this, 'add_warning_to_plugin_meta' ], 10, 2 );
+	}
+
+	/**
+	 * Get warning icon markup.
+	 *
+	 * @return string Warning icon markup.
+	 */
+	private function get_warning_icon() {
+		static $icon = null;
+		if ( null === $icon ) {
+			$icon = sprintf(
+				'<span style="font-family:%s">⚠️</span>',
+				esc_attr( Fonts::getEmojiFontFamilyValue() )
+			);
+		}
+		return $icon;
+	}
+
+	/**
+	 * Schedule the event.
+	 *
+	 * This does nothing if the event is already scheduled.
+	 *
+	 * @return void
+	 */
+	public function schedule_event() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		$event_name = $this->get_event_name();
+		$timestamp  = wp_next_scheduled( $event_name );
+
+		if ( $timestamp ) {
+			return;
+		}
+
+		wp_schedule_event( time(), $this->get_interval(), $event_name );
+	}
+
+	/**
+	 * Run deactivation logic.
+	 *
+	 * This should be hooked up to the WordPress deactivation hook.
+	 *
+	 * @todo Needs refactoring if used for more than one cron-based task, to avoid iterating over sites multiple times.
+	 *
+	 * @param bool $network_wide Whether the deactivation was done network-wide.
+	 * @return void
+	 */
+	public function deactivate( $network_wide ) {
+		if ( $network_wide && is_multisite() && ! wp_is_large_network( 'sites' ) ) {
+			foreach ( get_sites(
+				[
+					'fields' => 'ids',
+					'number' => 0,
+				]
+			) as $blog_id ) {
+				switch_to_blog( $blog_id );
+				wp_clear_scheduled_hook( $this->get_event_name() );
+				restore_current_blog();
+			}
+		} else {
+			wp_clear_scheduled_hook( $this->get_event_name() );
+		}
+	}
+
+	/**
+	 * Add a warning sign to the network deactivate action on the network plugins screen.
+	 *
+	 * @param string[] $actions     An array of plugin action links. By default this can include 'activate',
+	 *                              'deactivate', and 'delete'.
+	 * @return string[]
+	 */
+	public function add_warning_sign_to_network_deactivate_action( $actions ) {
+		if ( ! wp_is_large_network() ) {
+			return $actions;
+		}
+
+		if ( ! array_key_exists( 'deactivate', $actions ) ) {
+			return $actions;
+		}
+
+		$actions['deactivate'] = preg_replace( '#(?=</a>)#i', ' ' . self::get_warning_icon(), $actions['deactivate'] );
```

Comment:
`get_warning_icon()` is being called statically although the function is not.

Reply:
Fixed in 9ad9b10f6/